components/brave_new_tab_ui/containers/newTab/settings/sponsoredImagesToggle.tsx

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+import * as React from 'react'
+import styled, { css } from 'styled-components'
+
+// Common components
+import { SettingsText } from '../../../components/default'
+import { Toggle } from '../../../components/toggle'
+
+// Assets
+import InfoIcon from '../../../components/default/settings/assets/info-icon.svg'
+
+// Utilities
+import { getLocale } from '../../../../common/locale'
+
+const isDarkTheme = (p: any) => {
+  return p.theme.name === 'Brave Dark'
+}
+
+interface Props {
+  onChange: () => void
+  onEnableRewards: () => void
+
+  checked: boolean
+  disabled: boolean
+  rewardEnabled: boolean
+  adsEnabled: boolean
+  canSupportAds: boolean
+}
+
+const Container = styled('div')<{}>`
```

Comment:
Very optional (entirely up to you):
```suggestion
const Container = styled.div`
```

Reply:
Cool! done.